Integrations/integration-Fortigate.yml

```diff
@@ -119,6 +119,26 @@ script:
             return addr_string
 
 
+    def bool_to_str(str_representing_bool):
+        return str_representing_bool and str_representing_bool.lower() == 'true'
+
+
+    def generate_src_or_dst_request_data(policy_id, policy_field, policy_field_value, keep_original_data, add_or_remove):
+        address_list_for_request = policy_field_value.split(",")
+        if bool_to_str(keep_original_data):
+            policy_data = get_policy_request(policy_id)[0] # the return value is an array with one element
+            existing_adresses_list = policy_data.get(policy_field)
+            existing_adresses_list = [address_data["name"] for address_data in existing_adresses_list]
+            if add_or_remove.lower() == "add":
+                for address in existing_adresses_list:
+                    if not address in address_list_for_request:
+                        address_list_for_request.append(address)
+            else:
+                address_list_for_request = [address for address in existing_adresses_list if address not in address_list_for_request]
```

Comment:
how is it different from lines 133-135

Reply:
Look again, these are two opposite actions:
On lines 133-135, I add every address from the existing addresses to the arg addresses if it isn't there yet.
The purpose of these lines is to make sure the updated source/destination includes all the existing addresses + the new ones.
On the other hand, on line 137 I'm taking the existing addresses and removing all the arg addresses.